cloudinit/kindadapter.go

```diff
@@ -27,37 +27,38 @@ import (
 	"sigs.k8s.io/kind/pkg/exec"
 )
 
-var supportedCloudCongfigActions = map[string]cloudCongfigActionBuilder{
-	"write_files": newWriteFilesAction,
-	"runcmd":      newRunCmdAction,
-}
+const (
+	// Supported cloud config modules
+	writefiles = "write_files"
+	runcmd = "runcmd"
+)
 
-type cloudCongfigActionBuilder func() cloudCongfigAction
+type actionFactory struct {}
+func (a *actionFactory) action(name string) (action, error) {
+	switch name {
+	case writefiles:
+		return newWriteFilesAction(), nil
+	case runcmd:
+		return newRunCmdAction(), nil
+	default:
+		return nil, errors.Errorf("cloud config module %q is not supported", name)
```

Comment:
I'm not sure how we are consuming this on the other side, but we might want to make sure that we do not hard fail in the case we see an unexpected module defined and rather just log and emit an event.

Reply:
I could see that. This could totally return a "named-no-op" action that we can log.